app/routes.py

```diff
@@ -0,0 +1,177 @@
+from flask import request
+from flask import redirect
+from werkzeug.exceptions import BadRequest
+
+from app import app
+from app.helpers import add_item
+from app.helpers import fetch_url
+from app.helpers.checks import check_params
+from app.helpers.response_generation import make_error_msg
+from app.models.dynamo_db import get_dynamodb_table
+from service_config import Config
+
+logger = logging.getLogger(__name__)
+
+base_response_headers = {
+    'Content-Type': 'application/json; charset=utf-8',
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Methods': 'GET, OPTION',
+    'Access-Control-Allow-Headers': 'Content-Type, Authorization, x-requested-with, Origin, Accept'
+}
+
+
+@app.route('/checker', methods=['GET'])
+def checker():
+    """
+    * Quick summary of the function *
+
+    Simple function to check if the service is alive with no call to any other functions.
+
+    * Abortions originating in this function *
+
+    None
+
+    * Abortions originating in functions called from this function *
+
+    None
+
+    * Parameters and return values *
+
+    :return: a simple json saying basically 'OK'
+    """
+    logger.info("Checker route entered at %f", str(time.time()))
+    response = make_response(jsonify({'success': True, 'message': 'OK'}), 200)
+    response.headers = base_response_headers
+    return response
+
+
+@app.route('/shorten', methods=['POST'])
+def create_shortlink():
+    """
+    * Quick summary of the function *
+
+    The create_shortlink route's goal is to take an url and create a shortlink to it.
+    The only parameter we should receive is the url to be shortened, within the post payload.
+
+    We extract the url to shorten, the request scheme, domain and base path and send them to a
+    function to check those parameters. (check_params)
+
+    If those parameters are checked, we create the response to be sent to the user and create the
+    shortlink to send back (add_item)
+
+    * Abortions originating in this function *
+
+    Abort with a 403 status code if the Origin header is not set nor one we expect.
+
+    * Abortions originating in functions called from this function *
+
+    Abort with a 400 status code from check_params, add_item
+    Abort with a 500 status code from add_item
+
+    * Parameters and return values *
+
+    :request: the request must contain a Origin Header, and a json payload with an url field
+    :return: a json in response which contains the url which will redirect to the initial url
+    """
+    logger.info("Shortlink Creation route entered at %f", time.time())
+    if request.headers.get('Origin') is None or not \
+            re.match(Config.allowed_domains_pattern, request.headers['Origin']):
+        logger.critical("Shortlink Error: Invalid Origin")
+        abort(make_error_msg(403, "Not Allowed"))
+    response_headers = base_response_headers
+    try:
+        url = request.json.get('url', None)
+    except BadRequest:
```

Comment:
Does really `request.json.get('url', None)` raise `BadRequest` ?

Reply:
Yes, it does. If there is no json given as a parameter, it raises by default a BadRequest. This is what perplexed me with some of my tests.